app/views/charge.njk

```diff
@@ -321,152 +321,161 @@
+                            name="addressCity"
+                            maxlength="100"
+                            class="govuk-input govuk-!-width-three-quarters"
+                            value="{{ addressCity }}"
+                            autocomplete="billing address-level2"/>
+                        </div>
+                        <div class="govuk-form-group{% if highlightErrorFields.addressPostcode %} govuk-form-group--error{% endif %} govuk-!-margin-bottom-0" data-validation="addressPostcode">
+                          <label id="address-postcode-lbl" for="address-postcode" class="govuk-label">
+                            <span
+                              class="address-postcode-label"
+                              data-label-replace="addressPostcode"
+                              data-original-label="{{ __p("cardDetails.postcode") }}">
+                              {{ __p("cardDetails.postcode") }}
+                            </span>
+
+                            {% if highlightErrorFields.addressPostcode %}
+                              <p class="govuk-error-message" id="error-address-postcode">
+                                {{ highlightErrorFields.addressPostcode }}
+                              </p>
+                            {% endif %}
+                          </label>
+                          <input id="address-postcode"
+                            type="text"
+                            name="addressPostcode"
+                            maxlength="10"
+                            class="govuk-input govuk-!-width-one-quarter"
+                            value="{{ addressPostcode }}"
+                            autocomplete="billing postal-code"/>
+                        </div>
+                    </fieldset>
+                  </div> 
                 {% endif %}
                 {% if (gatewayAccount.emailCollectionMode === 'MANDATORY')
                   or(gatewayAccount.emailCollectionMode === 'OPTIONAL') %}
-                  <div class="email-container govuk-!-width-three-quarters govuk-!-padding-top-4 govuk-!-margin-top-8 pay-!-border-top">
-                    <div class="govuk-form-group {% if highlightErrorFields.email %} govuk-form-group--error{% endif %}" data-validation="email">
-                      <legend for="email" class="govuk-!-margin-bottom-6">
-                        {% if allowApplePay or allowGooglePay %}
-                          <h3 class="govuk-heading-m govuk-!-margin-bottom-1 web-payment-button-heading">{{ __p("cardDetails.contactDetails") }}</h3>
-                        {% endif %}
-                        <h2 class="govuk-heading-m govuk-!-margin-bottom-1 non-web-payment-button-heading">{{ __p("cardDetails.contactDetails") }}</h2>
+                  <div class="govuk-!-width-three-quarters email-container govuk-!-padding-top-4 govuk-!-margin-top-8 pay-!-border-top">
+                    <fieldset class="govuk-fieldset" aria-describedby="email-hint">
```

Comment:
not sure why we have fieldset for email address?

Reply:
Good question - as you don’t usually have a `fieldset` for a single field.
We seem to have a `legend` and a hint for the `Email` field but no `fieldset` tag - which is incorrect.

Reply:
Just had another look - I think we should leave the legend tag in for now.
As it links the heading `Contact details` with the field `email`.

Also it seems like you can have a valid `fieldset` with just 1 field.
Please see this website:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset (edited)